src/ucp/proto/proto_select.inl

```diff
@@ -20,11 +28,78 @@ KHASH_IMPL(ucp_proto_select_hash, khint64_t, ucp_proto_select_elem_t, 1,
            kh_int64_hash_func, kh_int64_hash_equal)
 
 
+static UCS_F_ALWAYS_INLINE UCS_F_NON_NULL const ucp_proto_config_t*
+ucp_proto_thresholds_elem_get_config(const ucp_proto_threshold_elem_t *thresholds,
+                                     unsigned idx)
+{
+    return &thresholds[idx].proto_config;
+}
+
+static UCS_F_ALWAYS_INLINE const ucp_proto_threshold_elem_t*
+ucp_proto_thresholds_search(const ucp_proto_threshold_elem_t *thresholds,
+                            size_t msg_length)
+{
+#define UCP_PROTO_THRESHOLDS_CHECK(_arg, _i) \
+    if (ucs_likely(msg_length <= thresholds[_i].max_msg_length)) { \
+        return &thresholds[_i]; \
+    }
+
+    UCS_PP_FOREACH(UCP_PROTO_THRESHOLDS_CHECK, _, 0, 1, 2)
+#undef UCP_PROTO_THRESHOLDS_CHECK
+    return ucp_proto_thresholds_search_slow(thresholds + 3, msg_length);
+}
+
+static UCS_F_ALWAYS_INLINE uint8_t
+ucp_proto_select_op_attr_to_flags(uint32_t op_attr_mask)
```

Comment:
will it be used in next PRs?

Reply:
yes: https://github.com/yosefe/ucx/blob/wip/protov2/src/ucp/proto/proto_select.inl#L124